src/nunit.analyzers/WithinUsage/WithinUsageAnalyzer.cs

```diff
@@ -0,0 +1,118 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+using NUnit.Analyzers.Constants;
+using NUnit.Analyzers.Extensions;
+using NUnit.Analyzers.Helpers;
+
+namespace NUnit.Analyzers.WithinUsage
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class WithinUsageAnalyzer : BaseAssertionAnalyzer
+    {
+        private static readonly string[] SupportedIsMethods = new[]
+        {
+            NunitFrameworkConstants.NameOfIsEqualTo,
+            NunitFrameworkConstants.NameOfIsEquivalentTo
+        };
+
+        private static readonly DiagnosticDescriptor descriptor = DiagnosticDescriptorCreator.Create(
+            id: AnalyzerIdentifiers.WithinIncompatibleTypes,
+            title: WithinUsageAnalyzerConstants.Title,
+            messageFormat: WithinUsageAnalyzerConstants.Message,
+            category: Categories.Assertion,
+            defaultSeverity: DiagnosticSeverity.Warning,
+            description: WithinUsageAnalyzerConstants.Description);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(descriptor);
+
+        protected override void AnalyzeAssertInvocation(OperationAnalysisContext context, IInvocationOperation assertOperation)
+        {
+            if (!AssertHelper.TryGetActualAndConstraintOperations(assertOperation, out _, out var constraintExpression))
+            {
+                return;
+            }
+
+            foreach (var constraintPart in constraintExpression.ConstraintParts)
+            {
+                if (!SupportedIsMethods.Contains(constraintPart.GetConstraintName()))
+                    continue;
+
+                // e.g. Is.EqualTo(expected).Within(0.1)
+                // Need to check type of expected
+
+                var withinSuffix = constraintPart.GetSuffix(NunitFrameworkConstants.NameOfEqualConstraintWithin) as IInvocationOperation;
+
```

Comment:
Note that you don't allow for Is.EquivalentTo

Reply:
Noted!

Reply:
You will have to find a way to deal with the other use cases.